flexmeasures/data/models/annotations.py

```diff
@@ -41,6 +46,50 @@ class Annotation(db.Model):
     def duration(self) -> timedelta:
         return self.end - self.start
 
+    @classmethod
+    def save(
+        cls, df: pd.DataFrame, annotation_type: str, commit_session: bool = False
+    ) -> List["Annotation"]:
+        """Save a data frame with annotations.
+
+        Expects the following columns (or multi-index levels):
```

Comment:
Can you elaborate why this function should expect a DataFrame and not, say, a dict?

Reply:
To be honest, I just happened to be handling a DataFrame with annotations in our GripOnGas plugin. There, the choice for a DataFrame makes sense because of the slicing and sorting functionality of the timing columns. So here, I designed just what I needed, a function to save annotations that accepts input in the form I already had.

Reply:
My take after thinking about this again: These functions are not for saving finished Annotations (the best way then would be to pass `Annotation` objects), but to create and save *multiple* annotations from raw data (for which a data frame is a good representation).

There are method names to better represent this fact. I also looked up the similar method `TimedBelief.add` (which in similar fashion gets a `BeliefsDataFrame`).

I thus believe `Annotation.add` would be a better choice for a name here.

We can then (in this PR or in follow-ups, I'd add TODOs) also use similar parameters:

- expunge_session
- allow_overwrite
- bulk_save_objects
- commit_transaction (you have this, requires just a renaming)